pkg/kubectl/stop.go

```diff
@@ -355,7 +356,16 @@ func (reaper *StatefulSetReaper) Stop(namespace, name string, timeout time.Durat
 	}
 
 	errList := []error{}
-	for _, pod := range podList.Items {
+	for i := range podList.Items {
```

Comment:
You should be able to switch `StatefulSetHasDesiredReplicas` to use ObservedGeneration and use that instead of deleting the pods manually here. If you do that, separate PR.